conduit/Data/Conduit/List.hs

```diff
@@ -631,6 +633,25 @@ consumeC =
 {-# INLINE consumeC #-}
 STREAMING0(consume, consumeC, consumeS)
 
+-- | Group a stream into chunks of a given size. The last chunk may contain
+-- fewer than n elements.
+--
+-- Subject to fusion
+--
+-- Since ???
+chunksOf :: Monad m => Int -> Conduit a m [a]
+chunksOf n =
+    start
+  where
+    start = await >>= maybe (return ()) (loop n id)
```

Comment:
Instead of passing the `x` around below, how about:

```haskell
start = await >>= maybe (return ()) (\x -> loop n (x:))
```

You _could_ point-free that... but please don't, it'll hurt my head :)